api/v1alpha1/serviceintentions_webhook_test.go

```diff
@@ -398,3 +400,167 @@ func TestValidateServiceIntentions_Update(t *testing.T) {
 		})
 	}
 }
+
+// Test that we return patches to set Consul namespace fields to their defaults.
+// This test also tests OSS where we expect no patches since OSS has no
+// Consul namespaces.
+func TestHandle_ServiceIntentions_Patches(t *testing.T) {
+	otherNS := "other"
+
+	cases := map[string]struct {
+		newResource *ServiceIntentions
+		expPatches  []jsonpatch.Operation
+	}{
+		"all namespace fields set": {
+			newResource: &ServiceIntentions{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "foo-intention",
+					Namespace: "bar",
+				},
+				Spec: ServiceIntentionsSpec{
+					Destination: Destination{
+						Name:      "foo",
+						Namespace: "bar",
+					},
+					Sources: SourceIntentions{
+						{
+							Name:      "baz",
+							Namespace: "baz",
+							Action:    "allow",
+						},
+					},
+				},
+			},
+			expPatches: []jsonpatch.Operation{},
+		},
+		"destination.namespace empty": {
+			newResource: &ServiceIntentions{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "foo-intention",
+					Namespace: "bar",
+				},
+				Spec: ServiceIntentionsSpec{
```

Comment:
So this is kinda unrelated to the changes proposed here, but this test reminded me of it. Consul barfs if the len of []source is 0. We can add that validation as a separate PR and i dont want to block this, but if it isn't too much work, could we handle it in this PR? This is not a blocker on this PR.